optuna/samplers/_search_space/group_decomposed.py

```diff
@@ -0,0 +1,103 @@
+import copy
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from optuna.distributions import BaseDistribution
+from optuna.study import BaseStudy
+from optuna.trial import TrialState
+
+
+class _SearchSpaceGroup(object):
+    def __init__(self) -> None:
+        self._group: List[Dict[str, BaseDistribution]] = []
+
+    @property
+    def group(self) -> List[Dict[str, BaseDistribution]]:
+        return self._group
+
+    def add_distributions(self, distributions: Dict[str, BaseDistribution]) -> None:
+        self._group = _add_distributions(self.group, distributions)
+
+
+def _add_distributions(
+    group: List[Dict[str, BaseDistribution]], distributions: Dict[str, BaseDistribution]
+) -> List[Dict[str, BaseDistribution]]:
+    if len(distributions) == 0:
+        return group
+
+    for search_space in group:
+        _keys = set(search_space.keys())
```

Comment:
I don't see this a lot in Optuna but is there a specific meaning to prefixing local variables with underscores?

Reply:
Sorry for the confusion. There is no specific meaning to do that. I'll use the conventional variable names, such as `keys, dist_keys, intersection`.